crates/test/test-suite/src/tests/recover_client.rs

```diff
@@ -37,7 +26,8 @@ impl<A, B> Default for TestRecoverClient<A, B> {
 
 impl<Driver, A, B> TestCase<Driver> for TestRecoverClient<A, B>
 where
-    Driver: CanUseBinaryTestDriverMethods<A, B>,
+    Driver: CanUseBinaryTestDriverMethods<A, B>
+        + CanHandleRecoverClient<Driver::ChainDriverA, Driver::ChainA, Driver::ChainB>,
```

Comment:
why it is not `CanHandleRecoverClient<Driver::ChainDriverA, A, B>`?

Reply:
Because if this dependency https://github.com/informalsystems/hermes-sdk/pull/615/files#diff-edff5854ab72e40f676061003a2f06bae0847e6259159bf0667b2f4dbe488f0fR11 we would need to add additional restrictions to `A`. So specifying `Driver::ChainA` and `Driver::ChainB` avoid this issue as they both implement the `HasClientIdType`